modules/net-vpc-network-firewall/README.md

```diff
@@ -0,0 +1,233 @@
+# Google Cloud Network Firewall Policy
+
+This module allows creation and management of regional and global network firewall policy :
+
+Yaml abstraction for Firewall rules can simplify users onboarding and also makes rules definition simpler and clearer comparing to HCL.
+
+Nested folder structure for yaml configurations is optionally supported, which allows better and structured code management for multiple teams and environments.
+
+## Example
+
+### Terraform code
+
+```hcl
+module "global_policy" { # global firewall rules using YAML by defining rules file location
+  source             = "./fabric/modules/project"
+  global_policy_name = "global-policy-1"
+  project_id         = "my-project"
+  global_network     = "my-network"
+  deployment_scope   = "global"
+  data_folders       = ["./global-rules"]
+}
+
+module "global_policy_2" { # global firewall rules using HCL
+  source             = "./fabric/modules/project"
+  global_policy_name = "global-policy-1"
+  project_id         = "my-project"
+  deployment_scope   = "global"
+  firewall_rules = {
+
+      "rule-1" = {
+        action         = "allow"
+        description    = "rule 2"
+        direction      = "INGRESS"
+        disabled       = false
+        enable_logging = true
+        layer4_configs = [{
+          ports    = ["8080", "443", "80"]
+          protocol = "tcp"
+          },
+          {
+            ports    = ["53", "123-130"]
+            protocol = "udp"
+          }
+        ]
+        priority           = 150
+        src_ip_ranges      = ["0.0.0.0/0"]
+        target_secure_tags = ["516738215535", "839187618417"]
+      }
+
+      "rule-2" = {
+        action             = "allow"
+        description        = "rule 4"
+        direction          = "EGRESS"
+        disabled           = false
+        enable_logging     = true
+        target_secure_tags = ["516738215535"]
+        dest_ip_ranges     = ["192.168.0.0/24"]
+      }
+  }
+}
+
+module "regional_policy" { # regional firewall rules using YAML by defining rules file location
+  source                 = "./fw"
+  regional_policy_name   = "regional-policy-1"
+  project_id             = "my-project"
+  firewall_policy_region = "australia-southeast1"
+  deployment_scope       = "regional"
+  data_folders           = ["./regional-rules"]
+}
+# tftest skip
```

Comment:
Can you enable tests for this example?

Reply:
yup iv added the tftest section for my example but im not sure how to test it locally

Reply:
hi @juliocc iv created a test module in tests/modules/net_vpc_network_firewall

Reply:
Hi @juliocc iv created a test module for my code